source/protocol/lsp-types.adb

```diff
@@ -154,6 +160,45 @@ package body LSP.Types is
       return To_Unbounded_Wide_String (UTF_16);
    end To_LSP_String;
 
+   -------------------
+   -- To_LSP_String --
+   -------------------
+
+   function To_LSP_String
+     (Text : GNATCOLL.JSON.UTF8_Unbounded_String) return LSP_String
+   is
+      Res : LSP_String;
+      Len : constant Natural := Length (Text);
+      Current_Index : Positive := 1;
+   begin
+      loop
+         --  Process the decoding in chunks
+         declare
+            Bound : Natural := Natural'Min (Current_Index + Chunk_Size, Len);
+            Chunk : constant String (Current_Index .. Bound) := Slice
+              (Text, Current_Index, Bound);
+         begin
+            --  We don't want to cut a chunk in the middle of a long
+            --  character, so look at the last 4 bytes and cut before
+            --  any such long character, and cut if needs be.
+            if Bound /= Len then
+               for J in reverse 0 .. 3 loop
+                  if Character'Pos (Chunk (Chunk'Last - J)) > 16#7F# then
```

Comment:
I guess this is incorrect. Continuation codes are in range 0100_0000 .. 0111_1111. Others are valid bounds.